backend/src/helpers/secretScanning.ts

```diff
@@ -0,0 +1,437 @@
+
+  } else {
+    const {
+      ciphertext: encryptedSaltCiphertext,
+      iv: saltIV,
+      tag: saltTag
+    } = encryptSymmetric128BitHexKeyUTF8({
+      plaintext: salt,
+      key: encryptionKey
+    })
+
+    await new GitSecretBlindIndexData({
+      organizationId: organizationId,
+      encryptedSaltCiphertext,
+      saltIV,
+      saltTag,
+      algorithm: ALGORITHM_AES_256_GCM,
+      keyEncoding: ENCODING_SCHEME_UTF8
+    }).save()
+  }
+  return salt;
+};
+
+/**
+ * Check if there is already a Git secret blind index for the Git secret findings [gitSecrets]
+ * @param {Object} obj
+ * @param {String[]} obj.gitSecrets - raw Git secret values from the Infisical Radar scan
+ * @param {String} obj.salt - 16-byte random salt tied to the GitHub organization that is connected to Infisical Radar
+ * @returns {Promise<string[]>} - A Promise that resolves to the generated Git secret blind indexes.
+ */
+export const createGitSecretBlindIndexesWithSaltHelper = async ({
+  gitSecrets,
+  salt
+}: CreateGitSecretBlindIndexesWithSaltParams): Promise<string[]> => {
+
+  const gitSecretBlindIndexes: string[] = [];
+
+  for (const gitSecret of gitSecrets) {
+    const gitSecretBlindIndex = await createGitSecretBlindIndex({ gitSecret, salt })
+    gitSecretBlindIndexes.push(gitSecretBlindIndex);
+  }
+  return gitSecretBlindIndexes;
+};
+
+/**
+ * Generate blind index for the Git secret finding values [gitSecret]
+ * and salt [salt]
+ * @param {Object} obj
+ * @param {String} obj.gitSecret - raw Git secret value from the Infisical Radar scan
+ * @param {String} obj.salt - salted organization id for the GitHub organization connected to Infisical Radar
+ * @returns {Promise<string>} - A Promise that resolves to the generated Git secret blind index.
+ */
+export const createGitSecretBlindIndexWithSaltHelper = async ({
+  gitSecret,
+  salt
+}: CreateGitSecretBlindIndexWithSaltParams): Promise<string> => {
+
+  const gitSecretBlindIndex = await createGitSecretBlindIndex({gitSecret, salt})
+  return gitSecretBlindIndex;
+};
+
+/**
+ * Create encrypted Git secrets and corresponding blind indexes
+ * @param {Object} obj
+ * @param {Set<string>|string[]} obj.gitSecrets - raw values of the Git secrets found in the Infisical Radar scan
+ * @param {String[]} obj.gitSecretBlindIndexes - Git secret blind indexes
+ * @param {String} obj.organizationId - Infisical id for the GitHub organization connected to Infisical Radar
+ * @param {String} obj.salt - 16-byte random salt tied to the GitHub organization connected to Infisical Radar
+ * @returns {Promise<string[]>} - A Promise that resolves to the generated Git secret blind indexes.
+ */
+export const createGitSecretsHelper = async ({
+  gitSecrets,
+  gitSecretBlindIndexes,
+  organizationId,
+  salt,
+  status
+}: CreateGitSecretsParams): Promise<string[]> => {
+
+  const gitSecretsArray = Array.isArray(gitSecrets) ? gitSecrets : Array.from(gitSecrets); // change from set to array for the push event
```

Comment:
Will `gitSecrets` be something other than an array?

Reply:
yeah it will either be an array or a set depending on whether it's coming from the full repo scan or the push. The reason is because in the push we need to create the blind index on the fly and so we can just pass the unique Git secrets. the full repo scan needs to still create the blind indexes, only add the unique ones to the GitSecret schema but add all of them against the corresponding GitRisks

Reply:
why don't we convert the set to an array before it gets passed to this function? That way the parameters stay consistent